In [1]:
import pandas as pd

In [2]:
df=pd.read_csv(r"C:\Users\hp\Desktop\Indian Startup.csv")
df
#work on new data set---->  "Indian Startup"

Company/Brand  Founded Headquarters  \
0         CollegeDekho   2015.0      Gurgaon   
1                 BOX8   2012.0       Mumbai   
2                Simpl   2015.0    Bangalore   
3          8i Ventures   2018.0       Mumbai   
4            PayGlocal   2021.0    Bangalore   
...                ...      ...          ...   
1204          HobSpace   2019.0       Mumbai   
1205           DcodeAI   2020.0    New Delhi   
1206  Biocon Biologics   1978.0    Bangalore   
1207        KiranaKart   2020.0       Mumbai   
1208             Meddo   2018.0      Gurgaon   

                                Sector  \
0                           E-learning   
1                     Food & Beverages   
2                    Consumer Services   
3     Venture Capital & Private Equity   
4                   Financial Services   
...                                ...   
1204                            EdTech   
1205                            EdTech   
1206                     BioTechnology   
1207                        E-commerce   
1208                        Healthcare   

                                           What it does  \
0     Collegedekho.com is Student’s Partner, Friend ...   
1                      India's Largest Desi Meals Brand   
2     Simpl empowers merchants to build trusted rela...   
3     8i is a Mumbai & Bangalore based early stage f...   
4     PayGlocal is a FinTech solving for global paym...   
...                                                 ...   
1204  HobSpace is a platform for extra-curricular ac...   
1205       EMPOWER LEARNERS TO BE FUTURE READY USING AI   
1206  Biocon is a fully integrated pure play biosimi...   
1207  A better model for instant grocery delivery in...   
1208   Meddo is an end-to-end medical services provider   

                                           Founder/s  \
0                                       Ruchir Arora   
1                             Anshul Gupta, Amit Raj   
2                           Nitya, Chaitra Chidanand   
3                                     Vikram Chachra   
4     Prachi Dharani, Rohit Sukhija, Yogesh Lokhande   
...                                              ...   
1204                    Priya Goel Sheth, Harsh Jain   
1205                  Kartik Sharma, Ashish Aggarwal   
1206                             Kiran Mazumdar-Shaw   
1207                   Kaivalya Vohra, Aadit Palicha   
1208                                 Saurabh Kochhar   

                            Investor/s       Amount          Stage     Date   
0                     Disrupt ADQ, QIC  $35,000,000       Series B  Dec-2021  
1                         Tiger Global  $40,000,000            NaN  Dec-2021  
2          Valar Ventures, IA Ventures  $40,000,000       Series B  Dec-2021  
3                                  NaN  $50,000,000            NaN  Dec-2021  
4                Sequoia Capital India   $4,900,000       Series A  Dec-2021  
...                                ...          ...            ...       ...  
1204  Siddharth Bhaskar Shah, Upsparks     $100,000  Pre-series A1  Jan-2021  
1205                           S Chand     $500,000            NaN  Jan-2021  
1206       Goldman Sachs, Tata Capital  $70,000,000            NaN  Jan-2021  
1207           Global Founders Capital     $720,000       Pre-seed  Jan-2021  
1208  growX ventures, Venture Gurukool   $3,000,000   Pre-series A  Jan-2021  

[1209 rows x 10 columns]

In [10]:
#to observe data 
df.head()

Company/Brand  Founded Headquarters                            Sector  \
0  CollegeDekho   2015.0      Gurgaon                        E-learning   
1          BOX8   2012.0       Mumbai                  Food & Beverages   
2         Simpl   2015.0    Bangalore                 Consumer Services   
3   8i Ventures   2018.0       Mumbai  Venture Capital & Private Equity   
4     PayGlocal   2021.0    Bangalore                Financial Services   

                                        What it does  \
0  Collegedekho.com is Student’s Partner, Friend ...   
1                   India's Largest Desi Meals Brand   
2  Simpl empowers merchants to build trusted rela...   
3  8i is a Mumbai & Bangalore based early stage f...   
4  PayGlocal is a FinTech solving for global paym...   

                                        Founder/s  \
0                                    Ruchir Arora   
1                          Anshul Gupta, Amit Raj   
2                        Nitya, Chaitra Chidanand   
3                                  Vikram Chachra   
4  Prachi Dharani, Rohit Sukhija, Yogesh Lokhande   

                    Investor/s       Amount     Stage     Date   
0             Disrupt ADQ, QIC  $35,000,000  Series B  Dec-2021  
1                 Tiger Global  $40,000,000       NaN  Dec-2021  
2  Valar Ventures, IA Ventures  $40,000,000  Series B  Dec-2021  
3                          NaN  $50,000,000       NaN  Dec-2021  
4        Sequoia Capital India   $4,900,000  Series A  Dec-2021

In [11]:
df.tail()

Company/Brand  Founded Headquarters         Sector  \
1204          HobSpace   2019.0       Mumbai         EdTech   
1205           DcodeAI   2020.0    New Delhi         EdTech   
1206  Biocon Biologics   1978.0    Bangalore  BioTechnology   
1207        KiranaKart   2020.0       Mumbai     E-commerce   
1208             Meddo   2018.0      Gurgaon     Healthcare   

                                           What it does  \
1204  HobSpace is a platform for extra-curricular ac...   
1205       EMPOWER LEARNERS TO BE FUTURE READY USING AI   
1206  Biocon is a fully integrated pure play biosimi...   
1207  A better model for instant grocery delivery in...   
1208   Meddo is an end-to-end medical services provider   

                           Founder/s                        Investor/s  \
1204    Priya Goel Sheth, Harsh Jain  Siddharth Bhaskar Shah, Upsparks   
1205  Kartik Sharma, Ashish Aggarwal                           S Chand   
1206             Kiran Mazumdar-Shaw       Goldman Sachs, Tata Capital   
1207   Kaivalya Vohra, Aadit Palicha           Global Founders Capital   
1208                 Saurabh Kochhar  growX ventures, Venture Gurukool   

           Amount          Stage     Date   
1204     $100,000  Pre-series A1  Jan-2021  
1205     $500,000            NaN  Jan-2021  
1206  $70,000,000            NaN  Jan-2021  
1207     $720,000       Pre-seed  Jan-2021  
1208   $3,000,000   Pre-series A  Jan-2021

In [24]:
df.sample(5)

Company/Brand  Founded Headquarters              Sector  \
732          Capital Float   2013.0    Bangalore             FinTech   
786            MicroDegree   2019.0    Mangalore              EdTech   
145                 Planys   2015.0      Chennai            Maritime   
351               PayCardo   2021.0         Pune  Financial Services   
909  Optimized Electrotech   2017.0    Ahmadabad           Aeorspace   

                                          What it does  \
732  Capital Float is an online platform that provi...   
786  MicroDegree is an ed-tech platform working tow...   
145  Planys is an Indian deep tech start-up creatin...   
351  A true Contactless & Cashless and secure payme...   
909  Long distance, multispectral Surveillance Equi...   

                                             Founder/s  \
732               Gaurav Hinduja, Sashank Rishyasringa   
786      Gaurav Kamath, Rakesh Kothari, Manikanta Nair   
145  Tanuj Jhunjhunwala, Vineet Upadhyay, Rakesh Si...   
351                   Vivek Kumar Choubey, Vivek Kumar   
909  Anil Yekkala, Dharin Shah, Kuldeep Saxena, Pur...   

                                       Investor/s        Amount         Stage  \
732  Triodos Investment Management, SAIF Partners    $6,000,000           NaN   
786    Research Innovation Incubation Design Labs  $Undisclosed           NaN   
145           Kieretsu Forum, Lets Venture, Shell    $1,600,000  Pre-series A   
351                                      MaGEHold  $Undisclosed      Pre-seed   
909                                          GVFL    $1,800,000  Pre-series A   

        Date   
732  Jun-2021  
786  Jun-2021  
145  Nov-2021  
351  Sep-2021  
909  Apr-2021

# in the sample data we found irrelevant data "$Undisclosed" in amount field

In [5]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   Headquarters   1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What it does   1209 non-null   object 
 5   Founder/s      1205 non-null   object 
 6   Investor/s     1146 non-null   object 
 7   Amount         1202 non-null   object 
 8   Stage          781 non-null    object 
 9   Date           1209 non-null   object 
dtypes: float64(1), object(9)
memory usage: 94.6+ KB


In [ ]:
#object ------> catagorical data 
#float64------> numerical data
#problem in the data 
#1. data type of founded is numerical form it need to convert in to date form
#2. data type of amount is catagorical it need to convert in to numerical form 

In [6]:
df.corr()


Founded
Founded      1.0

In [ ]:
#the correlation function do not provide any relation in the data because the most 
#of data is catagorical 

In [7]:
df.describe()


Founded
count  1208.000000
mean   2016.655629
std       4.517364
min    1963.000000
25%    2015.000000
50%    2018.000000
75%    2020.000000
max    2021.000000

In [ ]:
#in the describe we  not get usefull information from the data because all the data 
#in the form of catagorical expect "founded"

In [8]:
df.isnull()

Company/Brand  Founded  Headquarters  Sector  What it does  Founder/s  \
0             False    False         False   False         False      False   
1             False    False         False   False         False      False   
2             False    False         False   False         False      False   
3             False    False         False   False         False      False   
4             False    False         False   False         False      False   
...             ...      ...           ...     ...           ...        ...   
1204          False    False         False   False         False      False   
1205          False    False         False   False         False      False   
1206          False    False         False   False         False      False   
1207          False    False         False   False         False      False   
1208          False    False         False   False         False      False   

      Investor/s  Amount  Stage  Date   
0          False   False  False  False  
1          False   False   True  False  
2          False   False  False  False  
3           True   False   True  False  
4          False   False  False  False  
...          ...     ...    ...    ...  
1204       False   False  False  False  
1205       False   False   True  False  
1206       False   False   True  False  
1207       False   False  False  False  
1208       False   False  False  False  

[1209 rows x 10 columns]

In [9]:
df.isnull().sum()


Company/Brand      0
Founded            1
Headquarters       1
Sector             0
What it does       0
Founder/s          4
Investor/s        63
Amount             7
Stage            428
Date               0
dtype: int64

In [ ]:
#"isnull()" with sum function gave the numerical count value of having null value in the
#data  then ve need to find the percentage of missing data 

In [8]:
#in founded and headquarters coloum only one data missing so we ignore it because it do
#not create any change in analysis.

In [10]:
#percent of missing data in Founder/s
a=df["Founder/s"].isnull().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of missing data in Founder/s= ",per)

percent of missing data in Founder/s=  0.3308519437551696


In [11]:
#percent of missing data in Founded
a=df["Founded"].isnull().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of missing data in Founded= ",per)

percent of missing data in Founded=  0.0827129859387924


In [12]:
#percent of missing data in Investor/s
a=df["Investor/s"].isnull().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of missing data in Investor/s = ",per)

percent of missing data in Investor/s =  5.2109181141439205


In [13]:
#percent of missing data in Amount
a=df["Amount"].isnull().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of missing data in Amount = ",per)

percent of missing data in Amount =  0.5789909015715468


In [14]:
#percent of missing data in Stage
a=df["Stage"].isnull().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of missing data in Stage= ",per)

percent of missing data in Stage=  35.401157981803145


In [16]:
#percent of total missing data in data set
a=df.isnull().sum().sum()
b=df.shape[0]
per=(a/b)*100
print("percent of total missing data in data set= ",per)

percent of total missing data in data set=  41.687344913151364


# percent of total missing data in data set =41.687344913151364

In [17]:
#problem in data set
#1. misssing data 
#2. amount is object type
#3. founded in float it must be in date formate
#4. amount is having irrelavent data 


In [15]:
#use location function to analise the data 
df.loc[df["Sector"]=="EdTech"]

Company/Brand  Founded Headquarters  Sector  \
186                BYJU'S   2011.0    Bangalore  EdTech   
192   Sunstone Eduversity   2011.0     Gurugram  EdTech   
254                Questt   2020.0     Gurugram  EdTech   
372             EventBeep   2019.0         Pune  EdTech   
382              Tinkerly   2015.0       Jaipur  EdTech   
...                   ...      ...          ...     ...   
1181             Pariksha   2015.0         Pune  EdTech   
1189          Practically   2018.0    Telangana  EdTech   
1195                  Kyt   2020.0    Bangalore  EdTech   
1204             HobSpace   2019.0       Mumbai  EdTech   
1205              DcodeAI   2020.0    New Delhi  EdTech   

                                           What it does  \
186   BYJU’S is India's largest ed-tech company and ...   
192   Sunstone Eduversity is a one-of-its-kind busin...   
254   Questt is a Free online homework app that lets...   
372   An inclusive community where students are insp...   
382   A STEM learning online-offline solutions provi...   
...                                                 ...   
1181  Pariksha - The Sucess App, India's largest ver...   
1189  Making Learning personal, immersive and experi...   
1195  An online-first, global academy for extracurri...   
1204  HobSpace is a platform for extra-curricular ac...   
1205       EMPOWER LEARNERS TO BE FUTURE READY USING AI   

                                              Founder/s  \
186                    Byju Raveendran, Divya Gokulnath   
192                        Ashish Munjal, Piyush Nangru   
254                        Akhil, Mohsin M, Rohit Pande   
372     Saurabh Mangrulkar, Rakhi Pal, Venkatesh Prasad   
382   O.P. Godara, Vivek Pathak, Kapil Arya, Sharad ...   
...                                                 ...   
1181  Deepak Choudhary, Karanvir Singh Shekhawat, Ut...   
1189                              Subbarao Siddabattula   
1195                        Bhavik Rathod, Tripti Ahuja   
1204                       Priya Goel Sheth, Harsh Jain   
1205                     Kartik Sharma, Ashish Aggarwal   

                                             Investor/s        Amount  \
186    Oxshott Capital Partners, XN Exponent, Edelweiss  $300,000,000   
192                                  WestBridge Capital   $28,000,000   
254                                     Celesta Capital    $6,750,000   
372   SucSEED Indovation Fund, Uincept, Palimala Ven...  $Undisclosed   
382                                   Navneet Education      $800,000   
...                                                 ...           ...   
1181      Venture Catalysts, 9Unicorns Accelerator Fund    $1,800,000   
1189                                      Siana Capital    $4,000,000   
1195                              Alpha Wave Incubation    $5,000,000   
1204                   Siddharth Bhaskar Shah, Upsparks      $100,000   
1205                                            S Chand      $500,000   

              Stage     Date   
186             NaN  Oct-2021  
192        Series B  Oct-2021  
254        Series A  Oct-2021  
372             NaN  Sep-2021  
382             NaN  Sep-2021  
...             ...       ...  
1181            NaN  Jan-2021  
1189   Pre-series B  Jan-2021  
1195       Series A  Jan-2021  
1204  Pre-series A1  Jan-2021  
1205            NaN  Jan-2021  

[102 rows x 10 columns]

# it gave all the data which have sector ="EdTech" 

# if we want to get top 3 rows in the sector data than we use slicing with location

In [17]:
df.loc[df["Sector"]=="EdTech"][0:3]

Company/Brand  Founded Headquarters  Sector  \
186               BYJU'S   2011.0    Bangalore  EdTech   
192  Sunstone Eduversity   2011.0     Gurugram  EdTech   
254               Questt   2020.0     Gurugram  EdTech   

                                          What it does  \
186  BYJU’S is India's largest ed-tech company and ...   
192  Sunstone Eduversity is a one-of-its-kind busin...   
254  Questt is a Free online homework app that lets...   

                            Founder/s  \
186  Byju Raveendran, Divya Gokulnath   
192      Ashish Munjal, Piyush Nangru   
254      Akhil, Mohsin M, Rohit Pande   

                                           Investor/s        Amount     Stage  \
186  Oxshott Capital Partners, XN Exponent, Edelweiss  $300,000,000       NaN   
192                                WestBridge Capital   $28,000,000  Series B   
254                                   Celesta Capital    $6,750,000  Series A   

        Date   
186  Oct-2021  
192  Oct-2021  
254  Oct-2021

In [18]:

df["Sector"].iloc[1:5]

1                    Food & Beverages
2                   Consumer Services
3    Venture Capital & Private Equity
4                  Financial Services
Name: Sector, dtype: object

In [3]:
#we can use "loc" and "iloc" function both at a time  
df.loc[df["Sector"]=="EdTech"].iloc[1:5]

Company/Brand  Founded Headquarters  Sector  \
192  Sunstone Eduversity   2011.0     Gurugram  EdTech   
254               Questt   2020.0     Gurugram  EdTech   
372            EventBeep   2019.0         Pune  EdTech   
382             Tinkerly   2015.0       Jaipur  EdTech   

                                          What it does  \
192  Sunstone Eduversity is a one-of-its-kind busin...   
254  Questt is a Free online homework app that lets...   
372  An inclusive community where students are insp...   
382  A STEM learning online-offline solutions provi...   

                                             Founder/s  \
192                       Ashish Munjal, Piyush Nangru   
254                       Akhil, Mohsin M, Rohit Pande   
372    Saurabh Mangrulkar, Rakhi Pal, Venkatesh Prasad   
382  O.P. Godara, Vivek Pathak, Kapil Arya, Sharad ...   

                                            Investor/s        Amount  \
192                                 WestBridge Capital   $28,000,000   
254                                    Celesta Capital    $6,750,000   
372  SucSEED Indovation Fund, Uincept, Palimala Ven...  $Undisclosed   
382                                  Navneet Education      $800,000   

        Stage     Date   
192  Series B  Oct-2021  
254  Series A  Oct-2021  
372       NaN  Sep-2021  
382       NaN  Sep-2021

In [32]:
#groupby

df.groupby(["Headquarters","Sector"]).mean().iloc[1:]

Founded
Headquarters       Sector                    
Ahmadabad          Aeorspace           2017.0
                   Consulting          2021.0
                   EdTech              2015.5
                   Rental              2014.0
Ahmedabad          Computer Software   2018.0
...                                       ...
The Nilgiris       Biotechnology       2016.0
Thiruvananthapuram Human Resources     2016.0
Trivandrum         E-learning          2021.0
Vadodara           Automotive          2018.0
West Bengal        EdTech              2019.0

[578 rows x 1 columns]

In [39]:
#find the sector which is located in gurgaon
df[df["Headquarters"]=="Gurgaon"].groupby(["Headquarters","Sector"]).mean()

Founded
Headquarters Sector                    
Gurgaon      B2B Manufacturing   2020.0
             E-learning          2015.0
             EdTech              2019.0
             FinTech             2018.0
             Gaming              2018.0
             Healthcare          2018.0
             Hospitality         2012.0